lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -1997,7 +1997,7 @@ namespace Js
                 // Also, if the object's type has not changed, we need to ensure that
                 // the cached property string for this property, if any, does not
                 // specify this object's type.
-                scriptContext->InvalidatePropertyStringCache(propId, obj->GetType());
+                scriptContext->InvalidatePropertyStringAndSymbolCaches(propId, obj->GetType());
```

Comment:
I still haven't found how to construct a test case where there's actually work to do here. Either the writable data detection bit is not set, the type was replaced with a new type, or the property string was not cached. To test that the cache was being invalidated, I temporarily removed the writable-data-only check.